File: app/src/v1/routers/empleados.py
from flask import Blueprint, jsonify, request
import uuid
from src.v1.db.conexion import conexion


empleados=Blueprint('empleados_blueprint',__name__)

@empleados.route('/agregar', methods=['POST'])
def agregar_empleado():
    try:
        data = request.json
        opcion = data.get('opcion')
        numeroempleado = data.get('numeroempleado')
        nombre = data.get('nombre')
        appaterno = data.get('appaterno')
        appmaterno = data.get('appmaterno')
        direccion = data.get('direccion')
        codigopostal = data.get('codigopostal')
        telefono = data.get('telefono')
        curp = data.get('curp')
        nss = data.get('nss')
        puesto = data.get('puesto')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionesempleados({opcion},{numeroempleado},'{nombre}','{appaterno}','{appmaterno}','{direccion}','{codigopostal}','{telefono}','{curp}','{nss}',{puesto},'');"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@empleados.route('/modificar', methods=['PUT'])
def modificar_empleado():
    try:
        data = request.json
        opcion = data.get('opcion')
        numeroempleado = data.get('numeroempleado')
        direccion = data.get('direccion')
        codigopostal = data.get('codigopostal')
        telefono = data.get('telefono')
        curp = data.get('curp')
        nss = data.get('nss')
        puesto = data.get('puesto')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionesempleados({opcion},{numeroempleado},'','','','{direccion}','{codigopostal}','{telefono}','{curp}','{nss}',{puesto},'');"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@empleados.route('/baja', methods=['DELETE'])
def baja_empleado():
    try:
        data = request.json
        opcion = data.get('opcion')
        numeroempleado = data.get('numeroempleado')
        causabaja = data.get('causabaja')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionesempleados({opcion},{numeroempleado},'','','','','','','','',0,'{causabaja}');"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

@empleados.route('/consulta', methods=['GET'])
def consulta_empleado():
    try:
        opcion = request.args.get('opcion')
        numeroempleado = request.args.get('numeroempleado')
        
        conn = conexion()
        conn.conectar()
        query = f"select * from fnoperacionesempleados({opcion},{numeroempleado},'','','','','','','','',0,'');"
        resultado = conn.ejecutarquery(query)
        conn.cerrar()
        return jsonify(resultado), 200
    except Exception as ex:
        return jsonify({'message': str(ex)}), 500

File: WWW/PHP/Clases/Empleados.php
<?php

declare(strict_types=1);

final class Empleados
{

    private static string $apiUrl = "http://api:3000/empleados";

    private function __construct()
    {
    }

    private static function callApi(string $method, string $endpoint, ?array $data = null): mixed
    {
        $url = self::$apiUrl . $endpoint;
        $ch = curl_init($url);

        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);

        if ($data) {
            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
            curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
        }

        $response = curl_exec($ch);
        curl_close($ch);

        return json_decode((string) $response, true);
    }

    public static function agregarEmpleado(int $opcion, int $numeroempleado, string $nombre, string $appaterno, string $appmaterno, string $direccion, string $codigopostal, string $telefono, string $curp, string $nss, int $puesto): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'numeroempleado' => $numeroempleado,
            'nombre' => $nombre,
            'appaterno' => $appaterno,
            'appmaterno' => $appmaterno,
            'direccion' => $direccion,
            'codigopostal' => $codigopostal,
            'telefono' => $telefono,
            'curp' => $curp,
            'nss' => $nss,
            'puesto' => $puesto
        );
        return self::callApi('POST', '/agregar', $data);
    }

    public static function modificarEmpleado(int $opcion, int $numeroempleado, string $direccion, string $codigopostal, string $telefono, string $curp, string $nss, int $puesto): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'numeroempleado' => $numeroempleado,
            'direccion' => $direccion,
            'codigopostal' => $codigopostal,
            'telefono' => $telefono,
            'curp' => $curp,
            'nss' => $nss,
            'puesto' => $puesto
        );
        return self::callApi('PUT', '/modificar', $data);
    }

    public static function bajaEmpleado(int $opcion, int $numeroempleado, string $causabaja): mixed
    {
        $data = array(
            'opcion' => $opcion,
            'numeroempleado' => $numeroempleado,
            'causabaja' => $causabaja
        );
        return self::callApi('DELETE', '/baja', $data);
    }

    public static function consultaEmpleadoActivo(int $opcion, int $numeroempleado): mixed
    {
        $endpoint = "/consulta?opcion={$opcion}&numeroempleado={$numeroempleado}";
        return self::callApi('GET', $endpoint);
    }

}